app.ts

```diff
@@ -0,0 +1,309 @@
+import { TableHeadingString } from "./classes/table-headings.js";
+import { RenderTableRows } from "./classes/render-rows.js";
+import { RenderTableHeading } from "./classes/render-headings.js";
+import { HasFormatMethod } from "./interfaces/has-format-method.js";
+
+let numOfRows = getNumOfRows();		// Call function to get number of rows to render based on current screen size
+let totalNumofRecords: number;		// Declare public variable to store total number of records from the server
+
+// Accessing form data from front end
+let fromIDElement = document.querySelector('#fromID') as HTMLParagraphElement;
+let toIDElement = document.querySelector('#toID') as HTMLParagraphElement;
+let numofrecords = document.querySelector('#numofrecords') as HTMLParagraphElement;
+
+// Initializing variables for first page of data
+let fromID = 1;
+let toID = numOfRows;
+fromIDElement.innerHTML = fromID.toString();
+toIDElement.innerHTML = toID.toString();
+
+// Create initial page with initial data
+createInitialPage(numOfRows);
+
+// Listen for change in window size
+let screenHeight = screen.height;
+window.addEventListener('resize', function(event) {
+	screenHeight = window.innerHeight;	
+	
+	// Change number of rows based on window size
+	if (screenHeight >= 750) {
+		numOfRows = 23;
+	} else if ((screenHeight < 750) && (screenHeight >= 550)) {
+		numOfRows = 18;
+	} else if ((screenHeight < 550) && (screenHeight >= 265)) {
+		numOfRows = 8;
+	} else if ((screenHeight < 265) && (screenHeight >= 95)) {
+		numOfRows = 3;
+	} else if (screenHeight < 95) {
+		numOfRows = 2;
+	}
+
+	// Recreate grid with new set number of rows
+	createInitialPage(numOfRows);
```

Comment:
I don't think your resize function should be calling `createInitialPage`. You should just be figuring out which rows to fetch, fetch them and then rebuild just the part of the table that needs to be changed.